config_validation_engine/validators/validators.py

```diff
@@ -1,12 +1,20 @@
 import yamale
 from yamale.validators import DefaultValidators, Validator
 from email.utils import parseaddr
+from .constraints import EmailDomain
 
 class Email(Validator):
     """Email Validator"""
+    constraints = [EmailDomain]
     tag = 'email'
     def _is_valid(self,value):
-        parseaddr(value)
+        if (str == type(value)):
```

Comment:
@pumpkinman008 can we use https://pypi.org/project/validate_email/ to validate if the email is a proper email string as well? 
Also, could you include the package in the requirements.txt in your next push.
Also, if you could squish all your commits into 1, that would be perfect.
Thanks again for your work on this. It looks pretty good already 👍

Reply:
@maany https://pypi.org/project/validate_email/ has a validate_email function which requires input value as a string.
It can be used as follows if only the type of value parameter is string. Else it throws a TypeError.
```
def _is_valid(self,value):
        if ((str == type(value)) & (validate_email(value) == True)):
            return True
        else:
            return False
```

Reply:
The code above looks good. 
I think the value parameter provided by yamale should by default be a string value. The email must be a string indeed as the following line matches it against a regex pattern: https://github.com/syrusakbary/validate_email/blob/master/validate_email.py#L127.